lib/test_longest_symbol_kunit.c

```diff
@@ -0,0 +1,42 @@
+// SPDX-License-Identifier: GPL-2.0
+/*
+ * Test the longest symbol length
+ */
+
+#include <kunit/test.h>
+#include <linux/stringify.h>
+#include <linux/kallsyms.h>
+
+#define DI(name) n##name##name
+#define DDI(name) DI(n##name##name)
+#define DDDI(name) DDI(n##name##name)
+#define DDDDI(name) DDDI(n##name##name)
+#define DDDDDI(name) DDDDI(n##name##name)
+
+/*Generate a symbol whose name length is 511 */
+#define LONGEST_SYM_NAME_LEN  DDDDDI(g1h2i3j4k5l6m7n)
+
+int LONGEST_SYM_NAME_LEN(void)
```

Comment:
Could we maybe mark this noinline, so we're actually loading another symbol.

It doesn't really matter, as this is all resolved at compile time anyway, but I'd feel weird about the whole test getting optimized down to nothing. :-)

Reply:
:+1: